### PR TITLE
v5.2.0 design refresh, plus responsive offcanvas classes

### LINKED ISSUE #33387: Responsive classes for offcanvas
I've got a sidebar down my page that I'd like to move into an offcanvas (looking forward to the full release of v5 to use that), but I'd like it to be "oncanvas" when the page is above a certain width (as a part of the page, rather than overlaid like a shown offcanvas). For an example, see the current version of Slack via the web - as you shrink the window the sidebar becomes an offcanvas element with a button to show it.

Slack-bar-oncanvas
> _Sidebar shown on the page_

Slack-bar-offcanvas
> _Sidebar moves to offcanvas, with toggle button in the header_

I'm imagining something like the existing responsive classes in Bootstrap would be a possible way of using this: `offcanvas`, `offcanvas-sm`, ..., `offcanvas-xxl`. When used with a breakpoint, the element would only have the offcanvas styles applied below the breakpoint.

### LINKED ISSUE #33993: Bug: scroll position changes when the button is dropdown
009455260bd0a7ab75fc0c9d2f295f7b

- Operating system and version: macOS, 11.3
- Browser and version: Safari, 14.1

### LINKED ISSUE #35594: Offcanvas navbar expand leaves stray overlay
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

A follow-up from https://github.com/twbs/bootstrap/issues/35577 - in the same scenario (an offcanvas navbar with `.navbar-expand-${breakpoint}`), if the offcanvas is opened, but then the browser window is resized, the offcanvas disappears but the semi-transparent overlay remains.

/cc @florianlacreuse 

### Reduced test cases

https://codepen.io/florianlacreuse/pen/wvreVxN

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

v5.1.3

### PR DESCRIPTION
**Pull request by mdo**
bootstrap-v520-home

bootstrap-v520-docs

This is a redesign of the homepage and docs layout to give v5.2.0 some new life. I've been playing around with this while recovering from my heart attack, so not everything is perfect, but it feels refreshing to me. And while the design has been updated, so too has the homepage content.

Here are the key changes:

- **New homepage design and content.** I've rewritten the homepage to better articulate what comes with Bootstrap and how it's powerful/useful to folks. Key content additions are the utility API, JavaScript plugins, and increasing CSS variable usage.

- **New documentation sidebar.** The sidebar of the docs pages has been updated to include colorful Bootstrap Icons as part of the section headers and an always expanded list of links. I've also written some JS (lol) to automatically scroll the sidebar to the `.active` link, so you can always tell where you are and quickly jump around that section.

- **Cross-linking page versions is coming.** I've added a first step at cross-linking the same page across the various versions, unless the page was added in another release (e.g., with the CSS Grid page in v5.1.0). Now in the dropdown, we'll give you a link back to each release and show a disabled version number if the page isn't a part of that previous release.

- **Adds a handful of new utilities.** Bootstrap now includes `.fw-semibold`, which we use across our new homepage, and additional `.border-*` sizes with `.border-4` and `.border-5`.

- **Refreshes `border-radius` on buttons and form controls.** You might notice that buttons, inputs, and more have been slightly tweaked with increased radii on their corners. This is purely an aesthetic change to make buttons look a little more button-y (and our form controls updated to match).

- **Adds responsive offcanvas support outside navbars.** Redesigns the navbar in our docs to use this to collapse our sidebar on narrow viewports. Fixes #33387.

There are some other miscellaneous updates to fine-tune things as well. I've fixed some inconsistent padding between docs navbar, subnav, content, and footer. I've also tweaked some page copy here and there as part of the refresh.

Fixes #33993 as well.

JS changes from @GeoSot fixes #35594.

---

### Todos

- [x] Consider reverting sidebar width change
- [x] Update the versions dropdown to include v5.1.3 links
- [ ] Consider cutting a v5.1.4 release with docs version picker changes? Or just push to `gh-pages`? _This would cross-link v5.1.x pages with their v5.0.x counterparts._
- [x] Double check any optimizations we might need (e.g., newly added custom icons sprite)
- [x] Document new `border` and `font-weight` utilities
- [x] Sidebar is too tall on mobile hurting usability since the user needs to scroll a lot
- [x] Work on JS and fix tests

Preview: https://deploy-preview-35736--twbs-bootstrap.netlify.app/

### CODE REVIEW

**Comment by patrickhlauke** [comment]
otherwise though, looking nice a modern and fresh. good stuff! will dig in a bit deeper later

File: site/assets/scss/_content.scss [hunk 1]
```diff
@@ -79,3 +79,35 @@
...
+  height: 1em;
+  fill: currentColor;
+}
+
+.icon-link {
+  display: flex;
+  align-items: center;
+  text-decoration-color: rgba($primary, .5);
+  text-underline-offset: .5rem;
+  -webkit-backface-visibility: hidden; // stylelint-disable-line property-no-vendor-prefix
```
Comment (XhmikosR) [inline site/assets/scss/_content.scss:94]: Maybe we should use the unprefixed property and let autoprefixer handle these?

**Comment by alecpl** [comment]
I don't like the radius change. `0.4rem` radius border looks "distorted" (more in Firefox, for some reason) for me.

**Comment by mdo** [comment]
> super nitpicky: with the page background now `#FEFFFF` rather than `#FFFFFF`, the blue links, pink/red code, in the prose now fall just below 4.5:1 contrast. it may be best to just bump the page background back to full white, rather than going with the "you failed by just a smidge" on the contrast.

The `body` `background-color` is still `#fff`.

> I don't like the radius change. `0.4rem` radius border looks "distorted" (more in Firefox, for some reason) for me.

Not seeing any distortion here.

**Comment by patrickhlauke** [comment]
> The `body` `background-color` is still `#fff`.

Oops, my fault, was going just by the screenshot rather than firing it up live (and oddly, the .png has a mix of `#ffffff`, `#efffff`, `#feffff` etc in the background, some compression artifact)

image

image

Anyway, ignore that then :)

File: site/content/docs/5.1/components/placeholders.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ title: Placeholders
 description: Use loading placeholders for your components or pages to indicate something may still be loading.
 group: components
 toc: true
+added: "5.1"
```
Comment (XhmikosR) [inline site/content/docs/5.1/components/placeholders.md:7]: Are these used?

File: site/data/plugins.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+- name: Alert
```
Comment (XhmikosR) [inline site/data/plugins.yml:1]: Couldn't we get this info via a front matter property and the page description? Just wondering if we really need a new file, not against it per se :)

File: site/layouts/_default/baseof.html [hunk 4]
```diff
@@ -5,6 +5,7 @@
   </head>
   {{ block "body_override" . }}<body>{{ end }}
     {{ partial "skippy" . }}
+    {{ partial "icons" . }}
```
Comment (XhmikosR) [inline site/layouts/_default/baseof.html:8]: Does the sprite include only the needed icons? If so, would be nice if we had a quick command documented to generate them in the future.

File: site/layouts/partials/favicons.html [hunk 5]
```diff
@@ -3,6 +3,6 @@
 <link rel="icon" href="/docs/{{ .Site.Params.docs_version }}/assets/img/favicons/favicon-32x32.png" sizes="32x32" type="image/png">
 <link rel="icon" href="/docs/{{ .Site.Params.docs_version }}/assets/img/favicons/favicon-16x16.png" sizes="16x16" type="image/png">
 <link rel="manifest" href="/docs/{{ .Site.Params.docs_version }}/assets/img/favicons/manifest.json">
-<link rel="mask-icon" href="/docs/{{ .Site.Params.docs_version }}/assets/img/favicons/safari-pinned-tab.svg" color="#7952b3">
+<link rel="mask-icon" href="/docs/{{ .Site.Params.docs_version }}/assets/img/favicons/safari-pinned-tab.svg" color="#712cf9">
```
Comment (XhmikosR) [inline site/layouts/partials/favicons.html:6]: Should we generate new favicons/social images too?

File: scss/_utilities.scss [hunk 6]
```diff
@@ -461,6 +461,7 @@ $utilities: map-merge(
         lighter: $font-weight-lighter,
         normal: $font-weight-normal,
         bold: $font-weight-bold,
+        semibold: $font-weight-semibold,
```
Comment (XhmikosR) [inline scss/_utilities.scss:464]: How about moving the new utilities to a separate PR and then rebase this one?

File: site/content/docs/5.1/components/placeholders.md (hunk 2 above)
Comment (mdo) [reply site/content/docs/5.1/components/placeholders.md:7]: This frontmatter tells the version picker in the subnav that if you're browsing v5.1.x and you want to see the placeholders page in the previous version from that dropdown (v5.0.2), you cannot, because the page wasn't added until v5.1.

File: site/data/plugins.yml (hunk 3 above)
Comment (mdo) [reply site/data/plugins.yml:1]: Open to that as an optimization in the future. For now, keeping this as-is.

File: site/layouts/_default/baseof.html (hunk 4 above)
Comment (mdo) [reply site/layouts/_default/baseof.html:8]: I manually created this, copying and pasting and editing from the Bootstrap Icons. It does only include icons that are needed across the homepage and docs layouts. Not much is wasted between the two right now, and I'll likely add even more icons to the homepage in another commit.

File: scss/_utilities.scss (hunk 6 above)
Comment (mdo) [reply scss/_utilities.scss:464]: Rebased everything to isolate these changes to their own commit, I think that's okay since it's clearly outlined.

File: scss/_variables.scss [hunk 7]
```diff
@@ -488,9 +488,11 @@ $border-color:                $gray-300 !default;
 // scss-docs-end border-variables
 
 // scss-docs-start border-radius-variables
-$border-radius:               .25rem !default;
-$border-radius-sm:            .2rem !default;
-$border-radius-lg:            .3rem !default;
+$border-radius:               .4rem !default;
```
Comment (GeoSot) [inline scss/_variables.scss:491]: I am not sure, but this is the base variables file. Right? 
If we change the variables here, wouldn't be a breaking change, for dist files and for all that haven't override the radius variables?

File: site/assets/js/application.js [hunk 8]
```diff
@@ -119,7 +134,9 @@
 
   // Insert copy to clipboard button before .highlight
   var btnTitle = 'Copy to clipboard'
-  var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard">Copy</button></div>'
+  // var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard">Copy</button></div>'
+  var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard" title="Copy to clipboard"><svg class="bi" width="1em" height="1em" fill="currentColor"><use xlink:href="#clipboard"/></svg></button></div>'
```
Comment (GeoSot) [inline site/assets/js/application.js:138]:
```suggestion
  var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard" ><svg class="bi" width="1em" height="1em" fill="currentColor"><use xlink:href="#clipboard"/></svg></button></div>'
```
we can omit title as it is initialized on 147 line

File: site/assets/js/application.js [hunk 9]
```diff
@@ -144,13 +161,23 @@
...
 
     tooltipBtn.setContent({ '.tooltip-inner': 'Copied!' })
     event.trigger.addEventListener('hidden.bs.tooltip', function () {
       tooltipBtn.setContent({ '.tooltip-inner': btnTitle })
     }, { once: true })
     event.clearSelection()
+    iconFirstChild.setAttributeNS(namespace, 'href', originalXhref.replace('clipboard', 'check2'))
+
+    setTimeout(function () {
+      iconFirstChild.setAttributeNS(namespace, 'href', originalXhref)
```
Comment (GeoSot) [inline site/assets/js/application.js:178]: `setTimeout` maybe could be avoided, if we put the code-block on line 172 (inside hidden tooltip event)

File: scss/_variables.scss (hunk 7 above)
Comment (mdo) [reply scss/_variables.scss:491]: I'm changing the default values, but I don't see this as a breaking change. Going for a visual refresh on a few components alongside our docs.

File: site/assets/js/application.js (hunk 9 above)
Comment (mdo) [reply site/assets/js/application.js:178]: Thanks, just tried that. While it works, I prefer the not-so-instant change of the icon and tooltip. It feels a little jarring without the timeout, so I'm going to leave it as-is for now :).

File: site/assets/js/application.js [hunk 10]
```diff
@@ -119,7 +134,9 @@
 
   // Insert copy to clipboard button before .highlight
   var btnTitle = 'Copy to clipboard'
-  var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard">Copy</button></div>'
+  // var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard">Copy</button></div>'
+  var btnHtml = '<div class="bd-clipboard"><button type="button" class="btn-clipboard" ><svg class="bi" width="1em" height="1em" fill="currentColor"><use xlink:href="#clipboard"/></svg></button></div>'
```
Comment (ffoodd) [inline site/assets/js/application.js:138]: This needs some accessibility love: I'd go with `role="img" aria-label="Copy"` on the SVG, just in case tooltip doesn't work or something.

And feel free to drop the previous line instead of commenting, I thought it was some docs :D

File: site/assets/scss/_component-examples.scss [hunk 11]
```diff
@@ -330,6 +330,11 @@
     white-space: pre;
     background-color: transparent;
     border: 0;
+
+    // Undo tabindex that's automatically added by Hugo
+    &:focus {
+      outline: 0;
```
Comment (ffoodd) [inline site/assets/scss/_component-examples.scss:336]: Consider using `outline-color: transparent` instead: that'd allow OS' high constrasts modes to better operate.

File: site/assets/scss/_masthead.scss [hunk 12]
```diff
@@ -7,32 +7,70 @@
...
+    @include border-radius(.5rem);
+  }
+  .btn-clipboard {
+    top: .5rem;
+  }
+
+  #carbonads { // stylelint-disable-line selector-max-id
+    margin-right: auto;
+    margin-left: auto;
+  }
```
Comment (ffoodd) [inline site/assets/scss/_masthead.scss:34]: Could we imagine using utilities for such things? `.mx-auto` should be fine, isn't it? And there're quite a few declarations in site styles that we may drop. Could be tackled later, obviously.

File: site/assets/scss/_toc.scss [hunk 13]
```diff
@@ -40,3 +40,42 @@
...
+
+    &:focus,
+    &[aria-expanded="true"] {
+      box-shadow: 0 0 0 3px rgba(var(--bd-violet-rgb), .25);
+    }
+  }
+
+  @include media-breakpoint-up(md) {
+    pointer-events: none;
+  }
```
Comment (ffoodd) [inline site/assets/scss/_toc.scss:74]: I think the button should be hidden on large screens, and replaced by a heading or something: currently, when navigating with keyboard, I'm still able to activate the button, which makes the ToC flicks visually (and makes `aria-*` irrelevant).

File: site/layouts/_default/docs.html [hunk 14]
```diff
@@ -18,12 +18,18 @@ <h1 class="bd-title" id="content">{{ .Title | markdownify }}</h1>
 
       {{ if (eq .Page.Params.toc true) }}
         <div class="bd-toc mt-4 mb-5 my-md-0 ps-xl-3 mb-lg-5 text-muted">
-          <strong class="d-block h6 my-2 pb-2 border-bottom">On this page</strong>
-          {{ .TableOfContents }}
+          <button class="btn btn-link link-dark p-md-0 mb-2 mb-md-0 fw-semibold text-decoration-none bd-toc-toggle" type="button" data-bs-toggle="collapse" data-bs-target="#tocContents" aria-expanded="false" aria-controls="tocContents">
```
Comment (ffoodd) [inline site/layouts/_default/docs.html:21]: Here it is: I'd go with `.d-lg-none` on the button itself, and another element (`p` or `hn` most likely) with `.d-none.d-lg-block`. WDYT?

File: site/layouts/_default/docs.html [hunk 15]
```diff
@@ -18,12 +18,18 @@ <h1 class="bd-title" id="content">{{ .Title | markdownify }}</h1>
...
+            On this page
+            <svg class="bi d-md-none ms-2"><use xlink:href="#chevron-expand"></use></svg>
```
Comment (ffoodd) [inline site/layouts/_default/docs.html:23]: Needs some accessibility love too: `aria-hidden="true"` on `svg` should be good.

File: site/layouts/partials/docs-sidebar.html [hunk 16]
```diff
@@ -16,26 +16,27 @@
     {{- $is_active_group := eq $.Page.Params.group $group_slug -}}
 
     {{- if $group.pages }}
-      <li class="mb-1">
-        <button class="btn d-inline-flex align-items-center rounded{{ if not $is_active_group }} collapsed{{ end }}" data-bs-toggle="collapse" data-bs-target="#{{ $group_slug }}-collapse" aria-expanded="{{ $is_active_group }}"{{ if $is_active_group }} aria-current="true"{{ end }}>
+      <li class="bd-links-group py-2">
+        <h6 class="bd-links-heading d-flex w-100 align-items-center">
```
Comment (ffoodd) [inline site/layouts/partials/docs-sidebar.html:20]: `h6` feels weird here, the headings hierarchy is quite messy: 
image
Maybe use a `<p class="h6">` instead?
@patrickhlauke Is it important or not that much? I remember your "When headings fails WCAG" post.

File: site/layouts/partials/docs-sidebar.html [hunk 17]
```diff
@@ -16,26 +16,27 @@
...
+          {{- if $group.icon }}
+            <svg class="bi me-2"{{- if $group.icon_color }} style="color: var(--bs-{{ $group.icon_color }});"{{- end }}><use xlink:href="#{{ $group.icon }}"></use></svg>
```
Comment (ffoodd) [inline site/layouts/partials/docs-sidebar.html:22]: Since it's decorative, it needs some `aria-hidden="true"` too :)

**Review by ffoodd — COMMENTED** [review]
Looks great, it's really refreshing to get some colors!

I noticed a few moves, mostly accessibility improvements.

As @alecpl, I'm seeing weird glitches with the new radii, here's what it looks like on my Firefox: 
image
I'm not sure what's happening here, maybe some subpixels rounding bug?

And regarding homepage, the first masthead get centered ("Get started anyway you want") which feels a bit strange since next blocks are left-aligned. I suppose it's to make things aligned with the main masthead?

The overall look & feel is very nice, I love it :)

File: site/assets/scss/_component-examples.scss (hunk 11 above)
Comment (mdo) [reply site/assets/scss/_component-examples.scss:336]: We use `outline: 0` in around a dozen other spots, maybe we need a more holistic look at that? Going to keep as-is because of that for now.

File: site/assets/scss/_masthead.scss (hunk 12 above)
Comment (mdo) [reply site/assets/scss/_masthead.scss:34]: All the HTML is added dynamically, so I cannot modify it with utilities.

File: site/assets/scss/_component-examples.scss (hunk 11 above)
Comment (ffoodd) [reply site/assets/scss/_component-examples.scss:336]: Seems legit, we'll address that everywhere at once (probably after your global focus vars PR).

File: site/layouts/partials/home/masthead-followup.html [hunk 18]
```diff
@@ -1,37 +1,39 @@
...
-    <div class="col-md-7 ps-md-5">
-      {{ highlight "npm install bootstrap" "sh" "" }}
-      {{ highlight (printf ("gem install bootstrap -v %s") .Site.Params.current_ruby_version) "sh" "" }}
+<div class="container masthead-followup">
+  <div class="col-lg-7 mx-auto pb-3 mb-3 mb-md-5 text-md-center">
+    <div class="masthead-followup-icon d-inline-block mb-3" style="color: var(--bs-indigo);">
+      <svg class="bi fs-1"><use xlink:href="#code"></use></svg>
     </div>
-  </section>
+    <h2 class="display-5 mb-3 fw-semibold lh-sm">Get started anyway you&nbsp;want</h2>
```
Comment (rabuckley) [inline site/layouts/partials/home/masthead-followup.html:6]:
```suggestion
    <h2 class="display-5 mb-3 fw-semibold lh-sm">Get started any way you&nbsp;want</h2>
```
any way (in any of the ways) vs anyway (regardless/nonetheless)

**Comment by mdo** [comment]
Thanks @ffoodd, changes pushed. I adjusted the default `border-radius` value from `.4rem` which calculates to `6.4px` to `.375rem` which comes out to an even 6px. That should help with those corners looking off.

**Comment by mdo** [comment]
Pushed some new changes to refine our navbar and its mobile behaviors.

- Navbar now has a slightly translucent gradient, making for a really cool effect IMO :).
- The search form and version picker has been moved to the main navbar, so we have no more need for the subnav which has now been removed. I'm wondering if we need to do something to upgrade to that new fangled Algolia search?
- I've also removed the Home link in the navbar because it's quite redundant next to the logo which links there.
- Lastly, I've moved the sidebar toggle to the navbar and switched up the icons for mobile views. TBD if I leave the `Browse` or similar word there for the docs sidebar menu toggle.

Have a look, let me know what you think—I'm really liking it.

Screen Shot 2022-02-20 at 10 04 20 AM

Screen Shot 2022-02-20 at 10 04 27 AM

**Comment by mdo** [comment]
### New design update

Aww yeah, I've got another huge update. This time I've added responsive offcanvas classes (modifying the navbar offcanvas styles in the process) and converted our sidebar into it's own offcanvas in the process.

new-docs-offcanvas-sidebar

### Responsive offcanvas

New responsive classes for the offcanvas are available with a new demo added into the docs.

responsive-offcanvas

responsive-offcanvas-open

### New Examples layout

I've also updated the Examples page to have a new layout.

new-docs-examples

File: site/layouts/_default/single.html [hunk 19]
```diff
@@ -1,14 +1,19 @@
...
         <div class="col-xl-8">
           <h1 class="bd-title mt-0">{{ .Title | markdownify }}</h1>
           <p class="bd-lead">{{ .Page.Params.Description | markdownify }}</p>
           {{ if eq .Title "Examples" }}
-          <div class="d-flex flex-column flex-sm-row">
-            <a href="{{ .Site.Params.download.dist_examples }}" class="btn btn-lg btn-bd-primary" onclick="ga('send', 'event', 'Examples', 'Hero', 'Download Examples');">Download examples</a>
-            <a href="{{ .Site.Params.download.source }}" class="btn btn-lg btn-outline-secondary mt-3 mt-sm-0 ms-sm-3" onclick="ga('send', 'event', 'Examples', 'Hero', 'Download');">Download source code</a>
+          <div class="d-flex flex-column flex-md-row gap-3">
+            <a href="{{ .Site.Params.download.dist_examples }}" class="btn btn-lg bd-btn-lg btn-bd-primary d-flex align-items-center justify-content-center fw-semibold" onclick="ga('send', 'event', 'Examples', 'Hero', 'Download Examples');">
+              <svg class="bi me-2"><use xlink:href="#box-seam"></use></svg>
```
Comment (ffoodd) [inline site/layouts/_default/single.html:11]: Missing `aria-hidden="true"` here.
```suggestion
              <svg class="bi me-2" aria-hidden="true"><use xlink:href="#box-seam"></use></svg>
```

File: site/layouts/partials/docs-navbar.html [hunk 20]
```diff
@@ -1,74 +1,91 @@
...
+    <a class="navbar-brand p-0 me-0 me-lg-2" href="/" aria-label="Bootstrap">
       {{ partial "icons/bootstrap-white-fill.svg" (dict "class" "d-block my-1" "width" "40" "height" "32") }}
     </a>
 
-    <button class="navbar-toggler" type="button" data-bs-toggle="offcanvas" data-bs-target="#bdNavbar" aria-controls="bdNavbar" aria-expanded="false" aria-label="Toggle navigation">
-      {{ partial "icons/hamburger.svg" (dict "class" "bi" "width" "32" "height" "32") }}
-    </button>
+    <div class="d-flex d-lg-none order-3">
+      <button class="navbar-toggler p-2" type="button" data-bs-toggle="offcanvas" data-bs-target="#bdNavbar" aria-controls="bdNavbar" aria-expanded="false" aria-label="Toggle navigation">
+        <svg class="bi" width="24" height="24"><use xlink:href="#three-dots"></use></svg>
```
Comment (ffoodd) [inline site/layouts/partials/docs-navbar.html:18]:
```suggestion
        <svg class="bi" width="24" height="24" aria-hidden="true"><use xlink:href="#three-dots"></use></svg>
```

File: site/layouts/partials/home/masthead.html [hunk 21]
```diff
@@ -1,28 +1,33 @@
...
         </div>
-        <p class="text-muted mb-0">
-          Currently <strong>v{{ .Site.Params.current_version }}</strong>
-          <span class="px-1">&middot;</span>
-          <a href="https://getbootstrap.com/docs/4.6/getting-started/introduction/" class="link-secondary">v4.6.x docs</a>
-          <span class="px-1">&middot;</span>
-          <a href="/docs/versions/" class="link-secondary">All releases</a>
-        </p>
+        <a href="/docs/{{ .Site.Params.docs_version }}/getting-started/introduction/" class="btn btn-lg bd-btn-lg btn-bd-primary d-flex align-items-center justify-content-center fw-semibold" onclick="ga('send', 'event', 'Jumbotron actions', 'Get started', 'Get started');">
+          <svg class="bi me-2"><use xlink:href="#book-half"></use></svg>
```
Comment (ffoodd) [inline site/layouts/partials/home/masthead.html:17]:
```suggestion
          <svg class="bi me-2" aria-hidden="true"><use xlink:href="#book-half"></use></svg>
```

**Review by ffoodd — COMMENTED** [review]
I really like it! It's so refreshing.

Regarding offcanvas, I think we need some JS love for responsive variants: opening an offcanvas and resizing viewport until it's not applicable anymore successfully hide the content, but the backdrop remains.

Not quite sure we really need to handle resizing, but that's the only concern I have right now :)

May I suggest we also kind of update cheatsheets at the same time? Latest changes (like responsive offcanvas) should appear in here too, I guess; and the new header could fit really well with the current cheatsheet page, refreshing it too. Current design makes obvious I'm not a designer I guess ;)

The upcoming 5.2 will be awesome, thanks for the inspiration :heart:

**Comment by GeoSot** [comment]
>Regarding offcanvas, I think we need some JS love for responsive variants: opening an offcanvas and resizing viewport until it's not applicable anymore successfully hide the content, but the backdrop remains.

Did a small trick. If we are ok with this I can then add tests

**Comment by mdo** [comment]
@GeoSot I may have force-pushed over your commits at some point, I'm not sure. Any chance you can have another look?

**Comment by GeoSot** [comment]
@mdo can we keep the base class `offcanvas` in responsive offcanvas`es too? 
We may face some issues with js selectors

Example: `offcanvas offcanvas-lg`

**Comment by XhmikosR** [comment]
Just a quick comment, this PR introduces a lot of accessibility issues (contrast ratio, buttons etc).

We better fix the issues before landing it. Also, I still think the docs redesign should be untied from any utils addition since we've seen what this can cause from past PRs 😛

**Comment by mdo** [comment]
Cherry-picked out a few things into separate PRs for y'all to review and then I'll redo this PR to remove those pieces and give it a big update.

- #35979
- #35980
- #35981

Please review when you can :).

**Comment by patrickhlauke** [comment]
i've dropped the ball on reviewing this in much depth. let me know once it's stable enough, if there are still things getting reorganised/changed dramatically at this point. @XhmikosR could you share a list of concerns of what you've seen accessibility-wise, so I can make sure to check over those too?

on a completely silly point...i'm on a windows machine, so the mac command key ⌘ shortcut thing in the docs search seems a bit out of place...

docs search box, showing the ⌘K shortcut

**Comment by XhmikosR** [comment]
I'm using the bookmarklet from https://squizlabs.github.io/HTML_CodeSniffer/ and I see plenty of issues. This PR is now split into parts so some of the color issues can be seen in https://github.com/twbs/bootstrap/pull/35984

**Comment by patrickhlauke** [comment]
> I'm using the bookmarklet from https://squizlabs.github.io/HTML_CodeSniffer/ and I see plenty of issues. This PR is now split into parts so some of the color issues can be seen in #35984

took it for a quick spin. most of the warnings seem to be ignorable (where they relate to transparent background colours, where the bookmarklet is unable to do an automated contrast calculation - will need manual checking). the hard error about lack of submit button in the form is debatable/wrong i'd say. but yes, will need to do a proper concerted run-through of the changes (and also check contrast for all colour combos in dark mode)

**Comment by mdo** [comment]
The diff here has been dramatically updated with the removal of those other PRs mentioned in https://github.com/twbs/bootstrap/pull/35736#issuecomment-1063171257. Might be more I can pull out as I try to revisit the offcanvas changes that @GeoSot has suggested to me over Slack.

**Comment by mdo** [comment]
@patrickhlauke Shortcut for the search bar should be resolved now. Just tested it myself.

**Comment by patrickhlauke** [comment]
perhaps tweaking it even further to use `Ctrl+K` rather than `^K`? I'll admit that even I, as not exactly a noob, was initially a bit confused what the `^` meant

**Comment by mdo** [comment]
Just pushed some more changes, including bringing in DocSearch v3. Still some more to clean up—including an odd jumping when you hit `Esc` after the DocSearch dialog pops up.

Screen Shot 2022-04-13 at 5 31 32 PM

Screen Shot 2022-04-13 at 5 31 35 PM

**Comment by GeoSot** [comment]
@mdo nice catch on algolia v3. 
Can you please split it to another PR to check it in separate (It works but doesn't get results)?

**Comment by mdo** [comment]
Moved DocSearch changes to #36176.

**Review by GeoSot — APPROVED** [review]

**Comment by GeoSot** [comment]
It is very huge to be sure, but to the point I could follow, I didn't find something dysfunctional. 
So "nice job man" 🚀
